### PR TITLE
Fix inspector reporting wrong frames when using `DrawerLayoutAndroid` on the new arch

### PR DESCRIPTION
**Pull request by j-piasecki**
## Summary:

Due to `getContentOriginOffset` not being overridden by `AndroidDrawerLayoutShadowNode`, `findNodeAtPoint` function from `UIManagerBinding` was also taking into account the content of the drawer even if it was closed. This was also causing the frames always to be reported as if the drawer was on the left side.

This PR adds a custom shadow node and native state, so that `getContentOriginOffset` can return the correct values depending on the state of the drawer.

Fixes https://github.com/facebook/react-native/issues/44425

## Changelog:

[ANDROID] [FIXED] - Fixed layout inspector reporting wrong frames when using `DrawerLayoutAndroid` on the new arch

## Test Plan:

Tested on the reproducer from the issue.

### CODE REVIEW

File: packages/react-native/ReactCommon/react/renderer/components/drawerlayout/android/react/renderer/components/drawerlayout/AndroidDrawerLayoutState.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const Float containerWidth = 0;
+  const bool drawerOnLeft = true;
+  const bool drawerOpened = false;
+
+  folly::dynamic getDynamic() const;
+  MapBuffer getMapBuffer() const {
+    return MapBufferBuilder::EMPTY();
+  };
+
+#pragma mark - Getters
```
Comment (tomekzaw) [inline packages/react-native/ReactCommon/react/renderer/components/drawerlayout/android/react/renderer/components/drawerlayout/AndroidDrawerLayoutState.h:45]: There are no getters so maybe we can remove this mark?

Comment (j-piasecki) [reply packages/react-native/ReactCommon/react/renderer/components/drawerlayout/android/react/renderer/components/drawerlayout/AndroidDrawerLayoutState.h]: Done in 2407494b6357b5120895a0f0cc1c4d165b9b26c0)

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java [hunk 2]
```diff
@@ -109,6 +137,62 @@ public boolean onTouchEvent(MotionEvent ev) {
...
+
+    if (currentState.hasKey("containerWidth")) {
+      stateContainerWidth = currentState.getDouble("containerWidth");
+    }
+
+    if (currentState.hasKey("drawerWidth")) {
+      stateDrawerWidth = currentState.getDouble("drawerWidth");
+    }
+
+    if (isOpen != stateDrawerOpened || onLeft != stateOnLeft || Math.abs(stateContainerWidth - containerWidth) > delta  || Math.abs(stateDrawerWidth - drawerWidth) > delta) {
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java]: Removed in 2407494b6357b5120895a0f0cc1c4d165b9b26c0)

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java [hunk 3]
```diff
@@ -109,6 +137,62 @@ public boolean onTouchEvent(MotionEvent ev) {
...
+    updateState();
+  }
+
+  public void setStateWrapper(StateWrapper stateWrapper) {
+    this.mStateWrapper = stateWrapper;
+  }
+
+  public void updateState() {
+    this.updateState(mOpened, mDrawerPosition == Gravity.START, PixelUtil.toDIPFromPixel(getWidth()), PixelUtil.toDIPFromPixel(mDrawerWidth));
+  }
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java:152]: Done in 2407494b6357b5120895a0f0cc1c4d165b9b26c0)

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java [hunk 4]
```diff
@@ -109,6 +137,62 @@ public boolean onTouchEvent(MotionEvent ev) {
...
+      stateContainerWidth = currentState.getDouble("containerWidth");
+    }
+
+    if (currentState.hasKey("drawerWidth")) {
+      stateDrawerWidth = currentState.getDouble("drawerWidth");
+    }
+
+    if (isOpen != stateDrawerOpened || onLeft != stateOnLeft || Math.abs(stateContainerWidth - containerWidth) > delta  || Math.abs(stateDrawerWidth - drawerWidth) > delta) {
+      WritableNativeMap newState = new WritableNativeMap();
+      newState.putBoolean("drawerOnLeft", onLeft);
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/drawer/ReactDrawerLayout.java:189]: I'm not sure what's the convention, in other places plain strings are used as well: https://github.com/facebook/react-native/blob/362abb9ffea5af7e6e5616849fbce28166c758a0/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostView.kt#L467-L468
https://github.com/facebook/react-native/blob/362abb9ffea5af7e6e5616849fbce28166c758a0/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java#L1117-L1118

File: packages/react-native/ReactCommon/react/renderer/components/drawerlayout/android/react/renderer/components/drawerlayout/AndroidDrawerLayoutState.h [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        drawerOpened(data["drawerOpened"].getBool()) {};
+
+  const Float drawerWidth = 0;
+  const Float containerWidth = 0;
+  const bool drawerOnLeft = true;
+  const bool drawerOpened = false;
+
+  folly::dynamic getDynamic() const;
+  MapBuffer getMapBuffer() const {
+    return MapBufferBuilder::EMPTY();
```
[the remark this answers is not in the document]
Comment (j-piasecki) [reply packages/react-native/ReactCommon/react/renderer/components/drawerlayout/android/react/renderer/components/drawerlayout/AndroidDrawerLayoutState.h:42]: I believe it's done exactly the same way in modal component: https://github.com/facebook/react-native/blob/362abb9ffea5af7e6e5616849fbce28166c758a0/packages/react-native/ReactCommon/react/renderer/components/modal/ModalHostViewState.h#L44-L46

**Comment by j-piasecki** [comment]
Superseded by https://github.com/facebook/react-native/pull/44707

**Pull request closed without merging**